Speed_Controller/src/Timer/Timer.c

```diff
@@ -1,61 +1,82 @@
 #include "Timer.h"
 
-int8_t timeoutFlag = FALSE;
+#define TRUE 1
+#define FALSE 0
 
-/** Interrupt handler for Timer 2.
+static volatile int8_t timeoutFlag = FALSE;

```

Comment:
This needs to be volatile since it is set from an interrupt service routine. It has also been made static, and a getter and setter pair have been added for this flag to be used by main.c.